jcache/remote/src/test/java/org/infinispan/jcache/TrackingCacheEntryListener.java

```diff
@@ -0,0 +1,95 @@
+package org.infinispan.jcache;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.cache.event.CacheEntryCreatedListener;
+import javax.cache.event.CacheEntryEvent;
+import javax.cache.event.CacheEntryExpiredListener;
+import javax.cache.event.CacheEntryListenerException;
+import javax.cache.event.CacheEntryRemovedListener;
+import javax.cache.event.CacheEntryUpdatedListener;
+import javax.cache.event.EventType;
+
+import org.testng.Assert;
+
+/**
+ * Borrowed from TCK tests
+ *
+ * @author gustavonalle
+ */
+public class TrackingCacheEntryListener<K, V> implements CacheEntryCreatedListener<K, V>, CacheEntryUpdatedListener<K, V>, CacheEntryExpiredListener<K, V>, CacheEntryRemovedListener<K, V>, Serializable, AutoCloseable {
+   AtomicInteger created = new AtomicInteger();
+   AtomicInteger updated = new AtomicInteger();
+   AtomicInteger removed = new AtomicInteger();
+   ArrayList<CacheEntryEvent<K, V>> entries = new ArrayList<>();
```

Comment:
@wburns I'll make sure that all your comments about ```TrackingCacheEntryListener``` reach the TCK developers where I borrowed it from :smile:
Jokes apart, will fix them